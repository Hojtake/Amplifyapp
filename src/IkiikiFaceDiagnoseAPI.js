export default class IkiikiFaceDiagnoseAPI {
    async callLoginAPI (ID, password) {
        const myHeaders = new Headers();
        myHeaders.append("Content-Type", "application/json");
       
        const requestOptions = {
            method: 'POST',
            headers: myHeaders,
            body: JSON.stringify({ "ID": ID, "password": password }),
            redirect: 'follow'
        };
        const response = await fetch("https://nczxmo91g3.execute-api.ap-northeast-1.amazonaws.com/login", requestOptions);
        return response.json();
    }

    async callFaceDiagnoseAPI (Image, ID) {
        const myHeaders = new Headers();
        myHeaders.append("Content-Type", "application/json");
        const requestOptions = {
            method: 'POST',
            headers: myHeaders,
            body: JSON.stringify({ "Image": Image, "ID": ID }),
            redirect: 'follow'
        };
        const response = await fetch("https://sdnl1xmao1.execute-api.ap-northeast-1.amazonaws.com/ikiiki-value", requestOptions);
        return response.json();
    }
}