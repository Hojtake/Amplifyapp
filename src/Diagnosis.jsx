
import React from "react";
import ReactDOM from "react-dom";
import FunctionSelection  from "./FunctionSelection.jsx";
import IkiikiFaceDiagnoseAPI from "./IkiikiFaceDiagnoseAPI";
import classes from "./Diagnosis.module.css";
export default class Diagnosis extends React.Component {
    constructor(props){
        super(props);
        this.clickReturnToFunctionSelection = this.clickReturnToFunctionSelection.bind(this);
        this.clickDiagnose = this.clickDiagnose.bind(this);
        this.clickImageSelect = this.clickImageSelect.bind(this);
        this.state={resistDayMessage:null,resultMessage:null,operationMessage:"・画像選択ボタンを押してください",photoimage:""};
    }
    //画像をアップロードしたときに画像表示領域に画像を表示する関数
    clickImageSelect= (e)=>{
        
        this.setState({resultMessage:null});
        this.setState({resistDayMessage:null});
        const filelist = e.target.files;
        if(filelist.length == 0) return;
        this.setState({photoimage:""});
        const reader = new FileReader();
        const faceImage = new Image();
        const file = filelist[0];
        reader.onload = function (e) {
            faceImage.onload = function () {
                //画像表示領域の横幅(px)
                const windowWidth = 700;
                //画像表示領域の縦幅(px)
                const windowHeight =400;
                //画像表示領域の縦横比
                const windowRatio=windowHeight/windowWidth;
                //表示する画像の横幅
                let imageWidth;
                //表示する画像の縦幅
                let imageHeight;
                const ratio = faceImage.height / faceImage.width;
                //縦横比に応じて画面表示領域にフィットするようにimgのwidth,heightを決定する。
                if (ratio > windowRatio) {
                    //画像の縦幅を画像表示領域に合わせてフィットさせる
                    imageHeight = windowHeight;
                    imageWidth =faceImage.width * windowHeight / faceImage.height;
                }
                else {
                    //画像の横幅を画像表示領域に合わせてフィットさせる
                    imageHeight = faceImage.height * windowWidth / faceImage.width;
                    imageWidth=windowWidth;
                }
                //画像を画像表示領域内に埋め込む
                const imgdata = <img src={e.target.result} width={imageWidth} height={imageHeight} id="getimg"/>;
                this.setState({photoimage:imgdata})
            }.bind(this)
            faceImage.src= e.target.result;
        }.bind(this)
        reader.readAsDataURL(file);
        this.setState({operationMessage:"・診断するボタンを押してください"});
    }

    clickReturnToFunctionSelection = ()=>{
        ReactDOM.render(<FunctionSelection ID={this.props.ID}/>,document.getElementById("root"));
    }

    clickDiagnose = ()=>{
    	this.setState({resistDayMessage:null});
    	this.setState({resultMessage:null});
        const image = document.getElementById("getimg");
        if(image == null){
            this.setState({resultMessage:"画像を選択してから診断するボタンを押してください。"});
            return ;    
        }
        
        const api = new IkiikiFaceDiagnoseAPI();
        api.callFaceDiagnoseAPI(image.getAttribute("src"),this.props.ID)
        .then(result =>{
            //result及びresult内部のパラメータがnullまたはundefinedの場合にエラーとして処理を行う
            if(!result || !result.message){
                throw new Error();
                }
            if(result.hasFaceDiagnosed){
                if(!result.date || result.ikiikiValue === null || result.ikiikiValue === ""){
                    throw new Error();
                }
                this.setState({resistDayMessage:`${result.date}本日のイキイキ度は${result.ikiikiValue}です。`});
                this.setState({resultMessage:`${result.message}`});
            }else{                     
                this.setState({resultMessage:result.message});
            }                    
        }).catch(()=>{
            this.setState({resultMessage:"予期しないエラーが発生しました。しばらく待ってから再度実行してください。"});
        });           
    }

    render(){        
        return (
            <>
                <h1>イキイキ顔診断画面</h1>
                <div className={classes.username}><p id="ID">ID:{this.props.ID}</p></div>
                <div className={classes.return_to_function_select_area} ><button onClick={this.clickReturnToFunctionSelection} id="clickreturn">機能選択画面に戻る</button></div>
                <div className={classes.message_area}><p id="operationMsg">{this.state.operationMessage}</p></div>
                <div className={classes.photo_area} id="photo_area">{this.state.photoimage}</div>
                <div className={classes.button_area}>
                    <label htmlFor="filename" className={classes.label}>画像を選択<input type="file" id="filename" accept=".png,.jpg,.jpeg" onChange={this.clickImageSelect}/></label>    
                    <button className={classes.diagnose_button} onClick={this.clickDiagnose} id="diagnose_button">診断する</button>
                </div>
                <div className={classes.diagnose_result_area} id="diagnose_result_area">
                    <p id="resist_day">{this.state.resistDayMessage}</p>
                     <p id="result_msg">{this.state.resultMessage}</p>
                </div>
            </>
        );
    }
}