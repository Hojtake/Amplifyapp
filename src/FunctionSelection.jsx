import React from "react";
import ReactDOM from "react-dom";
import Diagnosis from "./Diagnosis";
import Login from "./Login";
import classes from"./FunctionSelection.module.css";

export default class FunctionSelection extends React.Component{
    constructor(props){
        super(props);
        this.clickLogout = this.clickLogout.bind(this);
        this.clickIkiikiDiagnoseFace = this.clickIkiikiDiagnoseFace.bind(this);
    }
    clickLogout = () =>{
        ReactDOM.render(<Login/>,document.getElementById("root"));
    }
    clickIkiikiDiagnoseFace = () => {
        ReactDOM.render(<Diagnosis ID={this.props.ID} ikiikiResults={this.props.ikiikiResults}/>,document.getElementById("root"));
    }

    render(){
        let errMsg=null;
        let ikiikiData=[]; 
        let noDataMsg= null
        if(this.props.hasReadData){
            if(this.props.ikiikiResults.length > 0){
                for(let i=0;i <this.props.ikiikiResults.length;i++){
                    let styleData;
                    let diagnosedData = this.props.ikiikiResults[i];
                    if(diagnosedData["ikiikiValue"]<50){
                        styleData="rgb(255, 125, 125)";
                    }else{
                        styleData="transparent";
                    }
                    ikiikiData.push(<tr key={i} style={{backgroundColor:styleData}}><td>{diagnosedData["diagnosedDate"]}</td><td>{diagnosedData["ikiikiValue"]}</td></tr>);
                
                }
            }else{
                noDataMsg = "まだイキイキ度が登録されていません。";
            }
        }
        return(
            <>
                <h1>機能選択画面</h1>
                <p className={classes.error_message}>{errMsg}</p>
                <div className={classes.username}><p>ID:{this.props.ID}</p></div>
                <div className={classes.logout_button}><button onClick={this.clickLogout} id="logout">ログアウト</button></div>
                <div className={classes.function_area}>
                <button className={classes.function} onClick={this.clickIkiikiDiagnoseFace} id="diagnoseFace">イキイキ顔診断</button>
                </div>
                <div className={classes.table}>
                <div className={classes.table_area}>
                <p>{this.props.ID}さんの過去のイキイキ度の記録</p>
                <p>{noDataMsg}</p>
                    <table border="2" width="650">
                    
                    <thead>
                    <tr>
                        <th width="50%">登録日</th>
                        <th>イキイキ度</th>
                    </tr>
                    </thead>
                    
                    <tbody>{ikiikiData}</tbody>
                    </table></div>
                    </div>
            </>
        )
    }

}