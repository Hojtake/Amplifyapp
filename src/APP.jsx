import React from "react";
import Login from "./Login";
import FunctionSelection from "./FunctionSelection";
import Diagnosis from "./Diagnosis";

export default class APP extends React.Component{
    constructor(props){
        super(props);
        this.state = {renderComponent : Login.name, ID : "", ikiikiResults:[],hasReadIkiikiResult:false};
        this.selectComponent = this.selectComponent.bind(this);
        this.setID = this.setID.bind(this);
        this.setHasReadIkiikiResult = this.setHasReadIkiikiResult.bind(this);
        this.setIkiikiResults = this.setIkiikiResults.bind(this);
    }
    
    selectComponent = (ComponentName) => {

        this.setState({renderComponent : ComponentName});
    }

    setID(ID){
        this.setState({ID : ID});
    }

    setHasReadIkiikiResult(result){
        this.setState({hasReadIkiikiResult: result});
    }

    setIkiikiResults(ikiikiResults){
        this.setState({ikiikiResults:ikiikiResults})
    }

    render(){
        if(this.state.renderComponent == Login.name){
            return(
                <>   
                    <Login 
                    callback={this.selectComponent} 
                    setID = {this.setID}
                    setIkiikiResults = {this.setIkiikiResults}
                    setHasReadIkiikiResult = {this.setHasReadIkiikiResult} />
                </>
            );
        }else if(this.state.renderComponent == FunctionSelection.name){
            return(
                <>   
                    <FunctionSelection 
                    callback={this.selectComponent} 
                    ID={this.state.ID} 
                    ikiikiResults={this.state.ikiikiResults}
                    hasReadIkiikiResult={this.state.hasReadIkiikiResult}
                    />
                </>
            );
        }else if(this.state.renderComponent == Diagnosis.name){
            return(
                <>   
                    <Diagnosis 
                    callback={this.selectComponent} 
                    ID={this.state.ID}
                    setIkiikiResults = {this.setIkiikiResults}
                    ikiikiResults={this.state.ikiikiResults}/>
                </>
            );
        }else{
            return(
                <>
                エラーが発生しました再度読み込みを行ってください
                </>
            )
        }
    }
}