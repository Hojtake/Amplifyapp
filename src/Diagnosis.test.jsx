import Diagnosis from "./Diagnosis";
import IkiikiFaceDiagnoseAPI from "./IkiikiFaceDiagnoseAPI";
import renderer from "react-test-renderer";
import { fireEvent, render, waitFor, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import fs from "fs";
import APP from "./APP";
import userEvent from "@testing-library/user-event";
import FunctionSelection from "./FunctionSelection";

jest.mock("./IkiikiFaceDiagnoseAPI");
const VALID_ID = "testuser";
const VALID_PASSWORD = "testpassword";
const dummyImageData = document.createElement("img");
dummyImageData.src = "dummy";
dummyImageData.id = "getimg";
//診断成功時の結果
const SUCCESS_RESULT_MSG = "本日も元気に働きましょう！";
const IKIIKI_VAL = 70;
const DATE = "2022/03/01";
const ikiikiResults =[        
    {"diagnoseDate":"2022/01/01","ikiikiValue":70 },
    {"diagnoseDate":"2022/01/02","ikiikiValue":50 },
    {"diagnoseDate":"2022/01/03","ikiikiValue":60 }];
const dummyResponseLoginResult = {
    hasLoginAuthenticated: true,
    ID: VALID_ID,
    message: null
}
const dummyResponseReadIkiikiResults = {
    hasReadIkiikiResult:true,
    ikiikiResults:[
        {"diagnoseDate":"2022/01/01","ikiikiValue":70 },
        {"diagnoseDate":"2022/01/02","ikiikiValue":50 },
        {"diagnoseDate":"2022/01/03","ikiikiValue":60 }
    ]
}

//想定しないエラーが発生したときのメッセージ
const UNEXPECTED_ERR_MSG = "予期しないエラーが発生しました。しばらく待ってから再度実行してください。";
//

it("正常系、イキイキ顔診断が成功した場合にメッセージが正しく表示されていることの確認", async () => {
    const dummyResponseJson
        = {
        hasFaceDiagnosed: true,
        ikiikiValue: IKIIKI_VAL,
        date: DATE,
        message: SUCCESS_RESULT_MSG
    }
    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson);
            }
        };
    });

    act(() => {
        render(<Diagnosis
            ID={VALID_ID}
            setIkiikiResults = {jest.fn()}
            ikiikiResults={ikiikiResults}
        />);
    });

    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        userEvent.click(diagnoseButton);
    });
    
    expect(document.querySelector("p[id='ID']").innerHTML).toBe(`ID:${VALID_ID}`);
    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(SUCCESS_RESULT_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe(`${DATE}本日のイキイキ度は${IKIIKI_VAL}です。`);
})

it("正常系、画像を選択したときにテスト操作ガイドのメッセージが更新されること", async () => {
    const imageFile = fs.readFileSync("./testImage/sample1.jpeg");

    const inputImage = new File([imageFile], "sample1.jpeg", { type: "image/jpeg", });
    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    const operationMsg = document.getElementById("operationMsg");
    expect(operationMsg.innerHTML).toBe("・画像選択ボタンを押してください");
    const inputFile = document.querySelector("input[type='file']");
    userEvent.upload(inputFile,inputImage);
    expect(inputFile.files[0]).toStrictEqual(inputImage);
    expect(inputFile.files).toHaveLength(1);
    expect(operationMsg.innerHTML).toBe("・診断するボタンを押してください");
});

it("正常系、機能選択画面に戻るボタンを押したときにFunctionSelectionがrenderされることの確認", async () => {
    act(() => {
        render(<APP/>);
    });

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callLoginAPI: () => {
                return Promise.resolve(dummyResponseLoginResult);
            },
            readIkiikiResultAPI: () => {
                return Promise.resolve(dummyResponseReadIkiikiResults);
            }
        };
    });
    const inputID = document.getElementById("ID");
    const inputPassWord = document.getElementById("password");
    userEvent.type(inputID, VALID_ID);
    userEvent.type(inputPassWord, VALID_PASSWORD);
    const loginButton = document.querySelector("button[type='submit']");
    await act(async () => {
        userEvent.click(loginButton);
    });
    const diagnoseFaceButton = document.querySelector("button[id='diagnoseFace']");
    await act(async () => {
        userEvent.click(diagnoseFaceButton);
    });
    const clickReturnToFunctionSelection = document.getElementById("clickreturn");
    await act(async () => {
        userEvent.click(clickReturnToFunctionSelection);
    });
    const tree = renderer.create(
        <FunctionSelection
        ID={VALID_ID}
        ikiikiResults={ikiikiResults}
        hasReadIkiikiResult={true}/>
    )
    expect(tree).toMatchSnapshot();
});

it("異常系、イキイキ顔診断が失敗した場合にメッセージが正しく表示されていることの確認", async () => {
    const FAILED_RESULT_MSG = "画像ファイルが大きすぎます。4MB以下の画像を選択してください。"
    const dummyResponseJson
        = {
        hasFaceDiagnosed: false,
        ikiikiValue: 0,
        date: null,
        message: FAILED_RESULT_MSG
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });

    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        userEvent.click(diagnoseButton);
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(FAILED_RESULT_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
})

it("異常系、レスポンスが正常に返らなかった場合のテスト（レスポンスの中身が空）", async () => {

    const dummyResponseJson
        = {

    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });

    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
})

it("異常系、レスポンスが正常に返らなかった場合のテスト（レスポンスの中身がnull）", async () => {
    const dummyResponseJson
        = null;
    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
})

it("異常系、診断は成功しているがイキイキ度がnullの場合のテスト", async () => {
    const dummyResponseJson
        = {
        hasFaceDiagnosed: true,
        ikiikiValue: null,
        date: DATE,
        message: SUCCESS_RESULT_MSG
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、診断は成功しているがイキイキ度が空白の場合のテスト", async () => {
    const dummyResponseJson
        = {
        hasFaceDiagnosed: true,
        ikiikiValue: "",
        date: DATE,
        message: SUCCESS_RESULT_MSG
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });

    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、診断は成功しているがdateがnullの場合のテスト", async () => {

    const dummyResponseJson
        = {
        hasFaceDiagnosed: true,
        ikiikiValue: IKIIKI_VAL,
        date: null,
        message: SUCCESS_RESULT_MSG
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、診断は成功しているがdateが空白の場合のテスト", async () => {

    const dummyResponseJson
        = {
        hasFaceDiagnosed: true,
        ikiikiValue: IKIIKI_VAL,
        date: "",
        message: SUCCESS_RESULT_MSG
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson);
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });

    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、診断は成功しているがmessageがnullの場合のテスト", async () => {
    const dummyResponseJson
        = {
        hasFaceDiagnosed: true,
        ikiikiValue: IKIIKI_VAL,
        date: DATE,
        message: null
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });

    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    
    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、診断は成功しているがmessageが空白の場合のテスト", async () => {

    const dummyResponseJson
        = {
        hasFaceDiagnosed: true,
        ikiikiValue: IKIIKI_VAL,
        date: DATE,
        message: ""
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、診断は成功しているがhasFaceDiagnosedがnullの場合のテスト", async () => {
    const dummyResponseJson
        = {
        hasFaceDiagnosed: null,
        ikiikiValue: IKIIKI_VAL,
        date: DATE,
        message: SUCCESS_RESULT_MSG
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、診断は成功しているがhasFaceDiagnosedが空白の場合のテスト", async () => {
    const dummyResponseJson
        = {
        hasFaceDiagnosed: "",
        ikiikiValue: IKIIKI_VAL,
        date: DATE,
        message: SUCCESS_RESULT_MSG
    }

    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callFaceDiagnoseAPI: () => {
                return Promise.resolve(dummyResponseJson
                );
            }
        };
    });

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    document.getElementById("photo_area").appendChild(dummyImageData);
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(UNEXPECTED_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");
});

it("異常系、画像を選択する前に診断を実行したの場合のテスト", async () => {
    const EMPTY_IMAGE_ERR_MSG = "画像を選択してから診断するボタンを押してください。"

    act(() => {
        render(<Diagnosis ID={VALID_ID} />);
    });
    const diagnoseButton = document.querySelector("button[id='diagnose_button']")
    await act(async () => {
        diagnoseButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
    });

    expect(document.querySelector("p[id='result_msg']").innerHTML).toBe(EMPTY_IMAGE_ERR_MSG);
    expect(document.querySelector("p[id='resist_day']").innerHTML).toBe("");

});

