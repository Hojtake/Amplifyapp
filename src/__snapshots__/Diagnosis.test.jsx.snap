// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`正常系、機能選択画面に戻るボタンを押したときにFunctionSelectionがrenderされることの確認 1`] = `
Array [
  <h1>
    機能選択画面
  </h1>,
  <p
    className="error_message"
    id="error_message"
  />,
  <div
    className="username"
  >
    <p>
      ID:
      testuser
    </p>
  </div>,
  <div
    className="logout_button"
  >
    <button
      id="logout"
      onClick={[Function]}
    >
      ログアウト
    </button>
  </div>,
  <div
    className="function_area"
  >
    <button
      className="function"
      id="diagnoseFace"
      onClick={[Function]}
    >
      イキイキ顔診断
    </button>
  </div>,
  <div
    className="table"
  >
    <div
      className="table_area"
    >
      <p
        id="table_title"
      >
        testuser
        さんの過去のイキイキ度の記録
      </p>
      <p
        id="nodata_message"
      />
      <table
        border="2"
        width="650"
      >
        <thead>
          <tr>
            <th
              width="50%"
            >
              登録日
            </th>
            <th>
              イキイキ度
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <td>
              2022/03/01
            </td>
            <td>
              70
            </td>
          </tr>
          <tr
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <td />
            <td>
              70
            </td>
          </tr>
          <tr
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <td />
            <td>
              50
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
